### DISCUSSION TITLE
Why is ActionPack enable_recognize_route disabled by default?

### DISCUSSION THREAD

**Discussion by richardmcmillen** [open-telemetry/opentelemetry-ruby#1379] [Q&A] [answered]
https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/main/instrumentation/action_pack/README.md#configuration-options

The ActionPack docs has a ominous sounding comment:

>The http.route attribute is disabled by default because we use .recognize

With a link to https://github.com/rails/rails/blob/v6.1.3/actionpack/lib/action_dispatch/journey/router.rb#L65

What is this comment trying to get at? Why is it disabled by default, is it for performance reasons? Or something else? It is a incredibly useful switch to enable but I am unsure of the consequences.

I would be willing to update the docs with whatever is discussed here to help others to!

**Comment by arielvalentin** [comment, marked as answer]
There is a related discussion here:

https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/123#discussion_r970942248

**Reply by richardmcmillen** [reply]
Thanks for responding and pointing this out. Though I am still unsure, why default the setting to false? Is it to reduce the computation? Is this a big enough issue to point out?

**Reply by richardmcmillen** [reply]
This comment does seem to give the impression that this option exists to allow opt-out of the potentially expensive method: https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/123#discussion_r994777255
